iron-api/src/main/java/io/axway/iron/Store.java

```diff
@@ -17,6 +17,15 @@
     void query(Consumer<ReadOnlyTransaction> storeQuery);
 
     /**
+     * Query the store data thanks a {@link ReadOnlyTransaction} and return a result.
+     *
+     * @param storeQuery the store query. The provided {@link ReadOnlyTransaction} must not be used outside the scope of the query method.
+     * @param <T> the return type of the query
+     * @return the result of the {@code storeQuery} function
+     */
+    <T> T queryWithResult(Function<ReadOnlyTransaction, T> storeQuery);
```

Comment:
i'm still not convinced by the name. Overloading with the query(Consumer) seems to have a small impact : only lambda that directly use an expression without {} body are ambiguous, but it's seems to be a non frequent use case in that context.